PhotoFeed/app/src/main/java/com/example/yhuang/photofeed/MainActivity.java

```diff
@@ -0,0 +1,212 @@
+
+        next.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty() && cur_page < items.size() - 1) {
+                    cur_page += 1;
+                    new ImageLoadTask(items.get(cur_page), imageview).execute();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Already the last page.", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        feed.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty()) {
+                    PhotoHelper.WriteIntoDatabase(mDbHelper, items.get(cur_page));
+                    Toast.makeText(getApplicationContext(), "Added to feed", Toast.LENGTH_SHORT).show();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Nothing to feed", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        delete.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (!items.isEmpty()) {
+                    PhotoHelper.DeleteItemDatabase(mDbHelper, items.get(cur_page));
+                    Toast.makeText(getApplicationContext(), "Deleted from feed", Toast.LENGTH_SHORT).show();
+                    openMyFeed();
+                } else {
+                    Toast.makeText(getApplicationContext(), "Nothing to delete", Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        // Handle presses on the action bar items
+        switch (item.getItemId()) {
+            case R.id.action_feed:
+                openMyFeed();
+                return true;
+            case R.id.action_search:
+                openSearch();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    /**
+     * the openSearch() method clears the current displaying data and opens the search page
+     */
+    public void openSearch() {
+        delete.setVisibility(View.GONE);
+        feed.setVisibility(View.VISIBLE);
+        editText.setVisibility(View.VISIBLE);
+        search.setVisibility(View.VISIBLE);
+        cur_page = 0;
+        items.clear();
+        imageview.setImageResource(0);
+    }
```

Comment:
Interesting non-fragment switching way of doing this. Thanks for letting us know about that ahead of time! As long as you understand how to do actual fragment switching this is perfect! @YuzhongHuang